server/controllers/mentionController.js

```diff
@@ -9,28 +8,50 @@ const Mention = require("../models/mentionModel");
  * @param {string} platforms - list of selected platforms seperated by coma
  * @param {string} keyword - keyword entered by user in UI
  * @param {string} sort - either "popularity" for most popular or "date" for most recent
+ * @param {number} page - page number for scrolling purposes
  */
 
 const getMentions = async (req, res) => {
   try {
-    const { platforms, keyword, sort } = req.query;
+    const { platforms, page, keyword, sort } = req.query;
     if (!platforms) {
       res.status(201).json({ mentions: [] });
     } else {
-      const platformsArray = platforms.split(",");
+      const platformsArray = platforms.split(',');
+      const sortOption = sort ? sort : 'date';
+      const dataPage = page ? parseInt(req.query.page) : 1;
+      const limit = 20;
+      const startIndex = (dataPage - 1) * limit;
+      const endIndex = dataPage * limit;
+      let nextPage;
+      let previousPage;
 
       // Fetching Mentions pertaining to selected platforms
-      const fetchMentions = (array) => {
-        const allMentions = [];
-        for (const platform of array) {
-          const mentions = Mention.find({ platform });
-          allMentions.push(mentions);
-        }
-        return allMentions;
+      // Sorting handled by MongoDB
+
+      const fetchMentions = async () => {
+        const getPlatformsObject = (array) => {
+          const platforms = [];
+          array.map((platform) => platforms.push({ platform }));
+          return platforms;
+        };
+
+        const getSortOption = (option) => {
+          if (option === 'date') {
+            return { date: -1 };
+          } else {
+            return { popularity: -1 };
+          }
+        };
+
+        const results = await Mention.find({
+          $or: [...getPlatformsObject(platformsArray)],
+        }).sort(getSortOption(sortOption));
+
+        return results;
```

Comment:
It is often (but not universally) a best practice to only nest functions that have side effects (calling B inside of A changes a variable scoped to A). In this case you could define these functions outside of the getMentions function (you would need to pass platformArray and sortOptions as args though). In that case it's generally preferred to organize them "top down", that is if A calls B and B calls C and then D, define them like this:
```javascript
const A = () => {}
const B = () => {}
const C = () => {}
const D = () => {}
```
Like I said this isn't a universal best practice as far as I know but is generally a good habit to be in for readable code.

Reply:
Thx Dan ! It's good to know that. I just moved the functions outside of getMentions.